python/src/uagents/agent.py

```diff
@@ -996,11 +1061,13 @@ async def run_async(self):
 
         """
         await self.setup()
+
+        tasks = [self._server.serve()]
+        if self._use_mailbox and self._mailbox_client is not None:
```

Comment:
Should we check for any registered REST handlers before running the server? Seems like this could save resources for a mailbox agent that has opted out of handling any rest endpoints, but not sure how significant.

Reply:
Good point. Two things come to mind though:
- we have to consider if the resource benefit is worth shutting down the server for an agent that only communicates via mailbox
- if the agent uses the inspector it will need a server anyway to provide the respective interfaces and currently we don't have an opt-out mechanism in place

Reply:
I refactored the code to:
- Allow the bureau to have agents with the same rest endpoints (if multiple endpoints are found it needs the "x-uagents-address" header to distinguish where to route to)
- Deactivate the server if mailbox is used and no rest endpoints are registered